### PR TITLE
DROOLS-1199 Memory leak in KieScanner

### PR DESCRIPTION
**Pull request by baldimir**
- Added reproducer for issue [1]. Issue is related to Plexus, because when using WiredComponentProvider instead of PlexusComponentProvider in MavenEmbedderUtils, issue is not there. 

[1] https://issues.jboss.org/browse/DROOLS-1199

### CODE REVIEW

File: kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java [hunk 1]
```diff
@@ -783,6 +787,82 @@ public void testKScannerWithGDRL() throws Exception {
...
+                averageMemoryFootprints.add(averageMemory);
+                averageMemory = 0;
+            }
+        }
+    }
+
+    private void waitForMillis(final long millis, final long startTimeMillis) {
+        while ((System.currentTimeMillis() - startTimeMillis) < millis) {
+            // do nothing - wait
+        }
```
Comment (yurloc) [inline kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java:853]: The busy wait is intended? Why not using Thread.sleep()?

Comment (baldimir) [inline kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java:853]: Yes it's intended. Thread.sleep() can be unreliable, especially on slow machines.

Comment (mrietveld) [inline kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java:853]: What about [Thread.sleep(long)](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#sleep%28long%29)??

Comment (baldimir) [inline kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java:853]: Please, see my previous comment.

**Comment by baldimir** [comment]
Closing this - migrated the test case to drools-test-coverage module with PR https://github.com/droolsjbpm/drools/pull/829

**Pull request closed without merging**